bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatHeaterHandler.java

```diff
@@ -0,0 +1,193 @@
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.millheat.internal.MillheatBindingConstants;
+import org.openhab.binding.millheat.internal.config.MillheatHeaterConfiguration;
+import org.openhab.binding.millheat.internal.model.Heater;
+import org.openhab.binding.millheat.internal.model.MillheatModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tec.uom.se.unit.Units;
+
+/**
+ * The {@link MillheatHeaterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arne Seime - Initial contribution
+ */
+@NonNullByDefault
+public class MillheatHeaterHandler extends MillheatBaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(MillheatHeaterHandler.class);
+    private @NonNullByDefault({}) MillheatHeaterConfiguration config;
+
+    public MillheatHeaterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        handleCommand(channelUID, command, getMillheatModel());
+    }
+
+    @Override
+    protected void handleCommand(ChannelUID channelUID, @NonNull Command command, @NonNull MillheatModel model) {
+        Optional<Heater> optionalHeater = model.findHeaterByMacOrId(config.macAddress, config.heaterId);
+        if (optionalHeater.isPresent()) {
+            Heater heater = optionalHeater.get();
+            if (MillheatBindingConstants.CHANNEL_CURRENT_TEMPERATURE.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    updateState(channelUID, new QuantityType<>(heater.getCurrentTemp(), Units.CELSIUS));
+                }
+            } else if (MillheatBindingConstants.CHANNEL_HEATING_ACTIVE.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    updateState(channelUID, heater.isHeatingActive() ? OnOffType.ON : OnOffType.OFF);
+                }
+            } else if (MillheatBindingConstants.CHANNEL_FAN_ACTIVE.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    updateState(channelUID, heater.isFanActive() ? OnOffType.ON : OnOffType.OFF);
+                } else if (heater.isCanChangeTemp() && heater.getRoom() == null) {
+                    updateIndependentHeaterProperties(null, null, command);
+                } else {
+                    logger.info("Heater {} cannot change temperature and is in a room", getThing().getUID());
```

Comment:
Should it be marked undef?

Reply:
Yes, any number would indicate actual power consumption